rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/db/RMDBManager.java

```diff
@@ -497,6 +497,24 @@ public Void doInTransaction(Session session) {
         }
     }
 
+    public void removeAllNodesFromNodeSource(final String nodeSourceName) {
+        try {
+            logger.info("Remove all nodes from node source " + nodeSourceName + IN_DATABASE_STRING);
+            executeReadWriteTransaction(new SessionWork<Void>() {
+                @Override
+                public Void doInTransaction(Session session) {
+                    session.getNamedQuery("deleteAllRMNodeDataFromNodeSource")
+                           .setParameter("name", nodeSourceName)
+                           .executeUpdate();
+                    return null;
+                }
+            });
+        } catch (RuntimeException e) {
+            throw new RuntimeException("Exception occurred while removing all nodes from node source " +
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)